code/modules/reagents/chemistry/holder/holder.dm

```diff
@@ -304,54 +304,42 @@
 	return total_removed //this should be amount unless the loop is prematurely broken, in which case it'll be lower. It shouldn't ever go OVER amount.
 
 /**
- * Removes all reagents by an amount equal to
- * [amount specified] / total volume present in this holder
+ * Removes all reagents either proportionally(amount is the direct volume to remove)
+ * when proportional the total volume of all reagents removed will equal to amount
+ * or relatively(amount is a percentile between 0->1) when relative amount is the %
+ * of each reagent to be removed
+ *
  * Arguments
  *
- * * amount - the volume of each reagent
+ * * amount - the amount to remove
+ * * relative - if TRUE amount is treated as an percentage between 0->1. If FALSE amount is the direct volume to remove
  */
-
-/datum/reagents/proc/remove_all(amount = 1)
+/datum/reagents/proc/remove_all(amount = 1, relative = FALSE)
 	if(!total_volume)
 		return FALSE
 
 	if(!IS_FINITE(amount))
 		stack_trace("non finite amount passed to remove all reagents [amount]")
 		return FALSE
+	if(relative && (amount < 0 || amount > 1))
+		stack_trace("illegal percentage value passed to remove all reagents [amount]")
+		return FALSE
```

Comment:
Why don't we just clamp the amount to 0 and 1 instead of throwing an error?

Reply:
Because we don't want to cover up people's mistakes. For e.g. we perform a calculation that yields bad values and pass it to this proc, by throwing a stack trace they can back track their steps with a debugger so they can correct their error outside this code rather than just covering it up and yield unexpected behaviour by clamping for them.